tests/test_schema.py

```diff
@@ -1,11 +1,33 @@
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import pytest
 import os.path
 
 from nomad.client import normalize_all, parse
 
 
+@pytest.mark.skip(
+    reason='These tests should only be for the schema validation and functionalities, not the actual data normalization and parsing'
+)
```

Comment:
I don't understand this. The normalization is here containing a lot of the functionality of the `ELNJupyterAnalysis`. Don't we need to test this @ka-sarthak?

Reply:
No, this is not the case.

Testing here should test functionalities of the classes defined in src. For now, it is only parsing, and I would say that yes, we need to extend this.

For now, I would skip it tho. It is not testing anything.

Reply:
I completely disagree. This might be how it was traditionally done but we are having a lot of *plugin functionallity* in the normalize function, such as creating the `.ipynb` file. This definitely needs to be tested to ensure that the plugin is working.

Reply:
This is not what the test is testing....

I don't refer to what we traditionally did or not. But if we want to do testing, ok for me, but this is not testing what you are refering to.

Reply:
We are testing that the normalization runs on line 31 `normalize_all(entry_archive)`... I agree that we should add more specific tests that the normalization actually yielded the results we expected. Also we need to add the `no_warn` fixture. @ka-sarthak please fill Chema in on what we discussed.

Reply:
This is not testing `normalize()` from the `ELNJupyterAnalysis` class. `normalize_all()` calls for the plugins defined in nomad/config, and as this one is a standalone package, it is not included there and hence not normalization is ran.

The test is just checking that `EntryArchive.data.analysis_type == 'test'`, which I understood was a simple dummy test added for later extend this.

Hence, I again said, **this is not testing anything from the plugin**.

Reply:
For the ELN classes (which are part of the `src/nomad_analysis/schema.py`) that generate the JupyterNotebooks and the metadata around it, we will need to test the normalization. To do this test `nomad.normalize_all` is needed (will be further extended to include more sensible tests). Testing the parser is, however, optional as I understand because we aren't really writing parsers. But using `nomad.parse` simplifies the creation of input needed for `nomad.normalize_all`. I was a bit unsure about this machinery before, sorry for that @JosePizarro3. But we will need `test_schema` to be part of the testing.

Something I almost forgot about, which @hampusnasstrom just reminded, is that `nomad.normalize_all` will not throw any errors to the `test_schema` even if something is wrong. This is because it handles the errors by adding them to the logger. As such the tests will always pass. 

We need to add a `no_warn` fixture that digs up for error statements in the logs and raise Assertions if errors are found.

Reply:
Again, you are not testing any normalization as of now because it will never run. The testing only constrains to checking that the key defined in `test.archive.yaml` is properly parsed.

In order to properly test the normalization, you must first import `ELNJupyterAnalysis` and, then, substitute `normalize_all(entry_archive)` by `ELNJupyterAnalysis.normalize(entry_archive)`. BUT, I doubt that this will run because:

1. You need to set the logger to None as default in each of your functions.
2. `entry_archive.m_context` might be None, and all the functionalities in ELNJupyterAnalysis depend on the m_context.

**This is exactly what I meant in my first comment to Hampus**, thus, and in order to stop losing time commenting on each other, this test should be skipped (for now).

Btw, everything you just replied, @ka-sarthak , implies touching functionalities from the core nomad package, which is what this testing should not be about.